src/main/java/net/pms/network/SearchRequestHandler.java

```diff
@@ -332,18 +334,19 @@ private int getDLNAResourceCountFromSQL(String query) {
 			try (Statement statement = connection.createStatement()) {
 				try (ResultSet resultSet = statement.executeQuery(query)) {
 					while (resultSet.next()) {
+						String filenameField = FilenameUtils.getBaseName(resultSet.getString("FILENAME"));
 						switch (type) {
 							case TYPE_ALBUM:
 							case TYPE_PERSON:
-								filesList.add(new VirtualFolderDbId(type, FilenameUtils.getBaseName(resultSet.getString("FILENAME")), ""));
+								filesList.add(new VirtualFolderDbId(filenameField, new DbidTypeAndIdent(type, filenameField), ""));
 								break;
 							case TYPE_PLAYLIST:
-								filesList.add(new VirtualFolderDbId(type, FilenameUtils.getBaseName(resultSet.getString("FILENAME")),
-									resultSet.getString("FID"), ""));
+								filesList
+									.add(new VirtualFolderDbId(filenameField, new DbidTypeAndIdent(type, resultSet.getString("FID")), ""));
```

Comment:
Here our convention would be like:
```suggestion
								filesList.add(
									new VirtualFolderDbId(
										filenameField,
										new DbidTypeAndIdent(type, resultSet.getString("FID")),
										""
									)
								);
```